api/src/main/java/edu/cornell/mannlib/vitro/webapp/application/VitroHomeDirectory.java

```diff
@@ -60,59 +71,209 @@ public String getDiscoveryMessage() {
+						// if the stored digest contains the file and its checksum
+						// equals existing file's checksum, it is ok to overwrite
+						write = storedDigest.containsKey(outFilename)
+							&& storedDigest.get(outFilename).equals(checksum(outFile));
+					}
+					if (write) {
+						outFile.getParentFile().mkdirs();
+						try (
+							InputStream is = new ByteArrayInputStream(bytes);
+							FileOutputStream fos = new FileOutputStream(outFile);
+						) {
+							IOUtils.copy(is, fos);
+						}
+					} else {
+						log.info(outFile.getAbsolutePath() + " changes have been preserved.");
+					}
+				}
+			}
+		} catch (IOException | NoSuchAlgorithmException e) {
+			throw new RuntimeException("Error creating home directory!", e);
+		}
+
+		return digest;
+	}
+
+	/**
+	 * Load checksum digest of VIVO home directory.
+	 * 
+	 * @return checksum digest
+	 */
+	private Map<String, String> loadDigest() {
+		File storedDigest = new File(getPath().toFile(), DIGEST_FILE_NAME);
+		if (storedDigest.exists() && storedDigest.isFile()) {
+			log.info("Reading VIVO home digest: " + storedDigest.getPath());
+			try {
+				return FileUtils.readLines(storedDigest, StandardCharsets.UTF_8)
+					.stream()
+					.map(this::split)
+					.collect(Collectors.toMap(this::checksumFile, this::checksumValue));
+			} catch (IOException e) {
+				throw new RuntimeException("Error reading home directory checksum digest!", e);
+			}
+		}
+		log.info("VIVO home digest not found: " + storedDigest.getPath());
+
+		return new HashMap<>();
+	}
+
+	/**
+	 * Write VIVO home checksum digest following md5 format; `<checksum> *<file>`.
+	 * 
+	 * @param digest checksum digest to write
+	 */
+	private void writeDigest(Map<String, String> digest) {
+		File storedDigest = new File(getPath().toFile(), DIGEST_FILE_NAME);
+		try (
+			FileOutputStream fos = new FileOutputStream(storedDigest); 
+			OutputStreamWriter osw = new OutputStreamWriter(fos);
+		) {
+			for (Map.Entry<String, String> entry : digest.entrySet()) {
+				String filename = entry.getKey();
+				String checksum = entry.getValue();
+				osw.write(String.format("%s *%s\n", checksum, filename));
+			}
+		} catch (IOException e) {
+			throw new RuntimeException("Error writing home directory checksum digest!", e);
+		}
+		log.info("VIVO home digest created: " + storedDigest.getPath());
+	}
+
+	/**
+	 * Split checksum.
+	 * 
+	 * @param checksum checksum delimited by space and asterisks `<checksum> *<file>`
+	 * @return split checksum
+	 */
+	private String[] split(String checksum) {
+		return checksum.split("\\s+");
+	}
+
+	/**
+	 * Get value from split checksum.
+	 * 
+	 * @param checksum split checksum
+	 * @return checksum value
+	 */
+	private String checksumValue(String[] checksum) {
+		return checksum[0];
+	}
+
+	/**
+	 * Return file from split checksum.
+	 * 
+	 * @param checksum split checksum
+	 * @return filename
+	 */
+	private String checksumFile(String[] checksum) {
+		return checksum[1].startsWith("*")
```

Comment:
It is an error is this is `false`, no?

Reply:
Yes, it could be. I was originally going to support only space delimiter.

Reply:
I think it best we validate these. Will had some checks.